web/config/index.ts

```diff
@@ -162,7 +162,7 @@ export const ANNOTATION_DEFAULT = {
   score_threshold: 0.9,
 }
 
-export const MAX_TOOLS_NUM = 10
+export const MAX_TOOLS_NUM = process.env.AGENT_MAX_TOOLS_NUM || 10
```

Comment:
This changes only works for the source code development, you can not get this via setting the environment variables.

https://github.com/langgenius/dify/blob/a4b2c10fb85effbb5dc932d6e40d164972737848/web/config/index.ts#L269